### DISCUSSION TITLE
Bunkrr

### DISCUSSION THREAD

**Discussion by Official-Husko** [Official-Husko/mul-tor#13] [Site Support]
Site URL: https://app.bunkrr.su/

99% of the support script is complete but for some reason it throws errors at me all the time. i can't figure out why so i will leave this to somebody else. It doesn't work in Insomnia or the code.

Detailed Issue: 
```json
{
    "description":"Invalid file UUID, or chunks data had already timed out. Try again?",
    "success":false
}
```

**Comment by dnice4040404** [comment]
> Site URL: https://app.bunkrr.su/
> 
> 99% of the support script is complete but for some reason it throws errors at me all the time. i can't figure out why so i will leave this to somebody else. It doesn't work in Insomnia or the code.
> 
> Detailed Issue:
> 
> ```json
> {
>     "description":"Invalid file UUID, or chunks data had already timed out. Try again?",
>     "success":false
> }
> ```

I was trying to build an uploader to bunkrr as well and this exactly where i'm stuck

**Reply by Official-Husko** [reply]
Perfect. Just fucking perfect. If you ever find a way to solve it, please let me know. Obviously, I will do the same if I can get it working.

**Reply by alexmi256** [reply]
What endpoint was failing for you two, the chunk uploading `/api/upload` or chunk finishing `/api/upload/finishchunks`?
I have not seen this error yet using aiohttp and `async with session.post("upload/finishchunks", json=upload_data) as resp:`. 
The tricky thing I ran into was that the server wanted `json` specifically and I was failing when using `async with session.post("upload/finishchunks", data=upload_data) as resp:` but all the other endpoints seem file with `data` instead of `json`.

**Reply by Official-Husko** [reply]
> What endpoint was failing for you two, the chunk uploading `/api/upload` or chunk finishing `/api/upload/finishchunks`? I have not seen this error yet using aiohttp and `async with session.post("upload/finishchunks", json=upload_data) as resp:`. The tricky thing I ran into was that the server wanted `json` specifically and I was failing when using `async with session.post("upload/finishchunks", data=upload_data) as resp:` but all the other endpoints seem file with `data` instead of `json`.

its the `/api/upload/finishchunks` for me. i provided a json with it and it still failed. Here is what it contained:

```json
{
	"files": [
		{
			"uuid": "4f514a29-27ae-4a19-a1ae-27ef3ea3732c",
			"original": "test.zip",
			"type": "application/x-zip-compressed",
			"albumid": "",
			"filelength": "",
			"age": ""
		}
	]
}
```

**Reply by alexmi256** [reply]
Just change `data` to `json` in the request params for finalize chunks and that will change the request type in the headers.
before: `raw_req = requests.post(url=finalize_url, data=upload_data, headers=headers)`
after: `raw_req = requests.post(url=finalize_url, json=upload_data, headers=headers)`

**Reply by Official-Husko** [reply]
@alexmi256 I do my testing with insomnia and in there it was set as json but it still never worked.

![image](https://github.com/Official-Husko/mul-tor/assets/27901510/e95b3023-acf5-46d5-bcec-93bf5f606806)

![image](https://github.com/Official-Husko/mul-tor/assets/27901510/97e9ffe5-65b9-4ec0-b4d9-1accd48d458c)
